R/object_name_linters.R

```diff
@@ -54,9 +54,12 @@ object_name_linter <- function(styles = c("snake_case", "symbols")) {
 
     generics <- strip_names(c(
       declared_s3_generics(xml),
-      namespace_imports(find_package(source_file$filename))$fun,
+      imported_s3_generics(namespace_imports(find_package(source_file$filename)))$fun,
       names(.knownS3Generics),
-      .S3PrimitiveGenerics, ls(baseenv())))
+      .S3PrimitiveGenerics,
+      base::.S3_methods_table[, 1L]
```

Comment:
This is much more restrictive than `ls(baseenv())` and captures all generics in `base` together with `.knownS3generics` and `.S3PrimitiveGenerics`.

cf. this:
```
is_generic <- vapply(ls_base, function(nm) is.function(get(nm, baseenv())) && isS3stdGeneric(get(nm, envir = baseenv())), logical(1L))
setdiff(names(which(is_generic)), c(.S3PrimitiveGenerics, .S3_methods_table[, 1], names(.knownS3Generics)))
# > character(0)
```

Reply:
Turns out this is incompatible with R < 3.5.0.
Added a fallback and factored the base generics out into a separate object `.base_s3_generics`.